### ISSUE TITLE
Nested Bookmarks with same name

### ISSUE THREAD

**Issue by elChapoSing** [py-pdf/PyPDF2#335]
When you try and nest several bookmarks with the same name, PyPdf2 does not take it into account. Below self-contained python code to test what I mean (you need at have 3 pdf files named a,b and c in the working folder to test it out)

```python
from PyPDF2 import PdfMerger, PdfReader

merger = PdfMerger()
pagenum = 0
first_one = True
for file in ["a.pdf", "b.pdf", "c.pdf"]:
    print("next row")
    reader = PdfReader(file)
    merger.append(reader)
    if first_one:
        child = merger.add_bookmark(title="blabla", pagenum=pagenum)
        first_one = False
    else:
        child = merger.add_bookmark(title="blabla", pagenum=pagenum, parent=child)

merger.write("test.pdf")
```

I would expect the resulting pdf to have three levels of nested bookmarks

    blabla
        blabla
            blabla

but instead I get

    blabla
        blabla
        blabla

PS: also posted on SO http://stackoverflow.com/questions/42941742/pypdf2-nested-bookmarks-with-same-name-not-working

**Comment by MartinThoma** [comment]
Thank you for reporting the issue :heart: 

 It seems as if this problem got solved in the past 5 years:

![image](https://user-images.githubusercontent.com/1658117/174474144-8b7e4fb7-9529-4788-963a-ab72d969e3be.png)

Please let me know if that is not the case

**Comment by MartinThoma** [comment]
I guess it was solved by #339
